### PR TITLE
Fix links

### PR DESCRIPTION
**Pull request by ocefpaf**
Should help fix the link-checker failure in #146.

@ayushanand18 I'm not sure if you are the one that created the bit.ly link but can you try to help me sort this out?
The link seems to redirect to an `http` page that doesn't seem correct to me. Also, it is broken in #146. Last but not least, I'd rather have the full link and skip code style there than have the info obfuscated by a URL shortener.

In this PR I added the full URL and the lint skips but I need some help to check if that URL is actually the one the authors of the docstrings added there originally.

### CODE REVIEW

File: pyobis/dataset/dataset.py [hunk 1]
```diff
@@ -34,8 +34,8 @@ def search(
     :param geometry: [String] Well Known Text (WKT). A WKT shape written as
         either POINT, LINESTRING, LINEARRING
         or POLYGON.
-        Example of a polygon: ((30.1 10.1, 20, 20 40, 40 40, 30.1 10.1)) would
-        be queried as http://bit.ly/1BzNwDq. Geometry, formatted as WKT or
+        Example of a polygon: ((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1)) would
+        be queried as https://api.gbif.org/v1/occurrence/search?geometry=POLYGON%28%2830.1+10.1%2C+10+20%2C+20+40%2C+40+40%2C+30.1+10.1%29%29. Geometry, formatted as WKT or
```
Comment (ocefpaf) [inline pyobis/dataset/dataset.py:38]: @ayushanand18 the polygon in the URL was `((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1))` . I fixed the text above this and used the full quoted URL below. However, that returns a 400 with the message:

```
Polygon with clockwise exterior ring: POLYGON((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1)). Did you mean these coordinates?  (Note this is only part of the polygon or multipolygon you provided.) LINEARRING (30.1 10.1, 40 40, 20 40, 10 20, 30.1 10.1)
```

I believe that either the example was never meant to work or something changed in the server recently.

File: pyobis/occurrences/occurrences.py [hunk 2]
```diff
@@ -251,8 +251,8 @@ def search(
     :param geometry: [String] Well Known Text (WKT). A WKT shape written
         as either POINT, LINESTRING, LINEARRING
         or POLYGON.
-        Example of a polygon: ((30.1 10.1, 20, 20 40, 40 40, 30.1 10.1)) would
-        be queried as http://bit.ly/1BzNwDq
+        Example of a polygon: ((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1)) would
+        be queried as https://api.gbif.org/v1/occurrence/search?geometry=POLYGON%28%2830.1+10.1%2C+10+20%2C+20+40%2C+40+40%2C+30.1+10.1%29%29
```
Comment (ayushanand18) [inline pyobis/occurrences/occurrences.py:255]: I believe, a valid OBIS URL pointing out to the correct use of `geometry` field would be - `https://api.obis.org/v3/occurrence?geometry=POLYGON%28%2830.1%2010.1%2C%2010%2020%2C%2020%2040%2C%2040%2040%2C%2030.1%2010.1%29%29`.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```
Comment (ayushanand18) [inline .github/workflows/tests.yml:10]: should we rather test only on even versions? something that works on alternate version should most probably work on the skipped ones.

Comment (ocefpaf) [reply .github/workflows/tests.yml:10]: I like to always test on oldest and latest at least. Even though you are correct, Python is known to introduce new features, deprecate others, and even break ABI in a micro release. (Ok, that last one happened only once but...)

TL;DR I'd wait to save CI time only/if we need in the future. For now, as long as the CIs are running OK, let's keep these. What do you think?

**Comment by ocefpaf** [comment]
> I'm very sorry for overlooking this, but it's most probably a tech debt (or what should I call it) carried forward from `pygbif` -> `pyobis` when original authors copied configs while creating the latter.

Not your fault. That is why we run tests ;-p

Thanks for the working URL. I'll update it accordingly.

**Comment by ayushanand18** [comment]
I hope this is ready to merge? just mentioning @ocefpaf for you to merge it whenever it's ready.

**Comment by ocefpaf** [comment]
> I hope this is ready to merge? just mentioning @ocefpaf for you to merge it whenever it's ready.

👍🏽 

> PS. Just to make sure we are not waiting for me.

I was but NP. I try to avoid self-merges as much as I can. Thanks for the review and for the fix!
